platforms/core-configuration/configuration-problems-base/src/main/kotlin/org/gradle/internal/cc/impl/problems/JsonWriter.kt

```diff
@@ -20,13 +20,24 @@ import com.fasterxml.jackson.core.JsonFactory
 import java.io.Writer
 
 class JsonWriter(private val writer: Writer) {
+
+    class JsonObject(val jsonWriter: JsonWriter) : AutoCloseable {
+        init {
+            jsonWriter.beginObject()
+        }
+
+        override fun close() {
+            jsonWriter.endObject()
+        }
+    }
+
     private
     val jsonGenerator = JsonFactory().createGenerator(writer)
 
     fun jsonObject(body: () -> Unit) {
-        beginObject()
-        body()
-        endObject()
+        JsonObject(this).use {
```

Comment:
Same as above

Reply:
some test broke because of an invalid state during JSON generation.
this guarantees a valid json for this cases.